### PR TITLE
Add gorilla mux instrumentation (instrumentation part)

### PR DESCRIPTION
**Pull request by andriikushch**
Instrumentation part of https://github.com/instana/go-sensor/pull/245

### CODE REVIEW

File: instrumentation/instamux/handler.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	instana "github.com/instana/go-sensor"
+)
+
+func AddMiddleware(sensor *instana.Sensor, router *mux.Router) {
+	router.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			pathTemplate, err := mux.CurrentRoute(r).GetPathTemplate()
+			if err != nil {
+				pathTemplate = ""
+			}
```
Comment (andrewslotin) [inline instrumentation/instamux/handler.go:21]: What could be the error here? Shall we add a debug log?

**Review by andrewslotin — APPROVED** [review]
LGTM! I've added a suggestion to log the path template retrieval error for your consideration

File: instrumentation/instamux/handler.go (hunk 1 above)
Comment (andriikushch) [reply instrumentation/instamux/handler.go:21]: There might be errors like when:
1. route does not define a path
2. when error resulted from building a route

I will add a logger, I do not see any reason not to have it. 👍

Comment (andriikushch) [reply instrumentation/instamux/handler.go]: @andrewslotin Here it is https://github.com/instana/go-sensor/pull/247/commits/b3d9874b1042d1bd65ee521b5599f1188084a0f8

**Pull request merged**